src/webview/web_context.rs

```diff
@@ -63,17 +71,32 @@ impl Default for WebContext {
   }
 }
 
+pub trait NavCallback: Fn(String, bool) -> bool + 'static {}
+
+impl<T: Fn(String, bool) -> bool + 'static> NavCallback for T {}
+
 /// Data that all [`WebContext`] share regardless of platform.
-#[derive(Debug, Default)]
+#[derive(Default)]
 pub struct WebContextData {
   data_directory: Option<PathBuf>,
+  nav_callback: Option<Arc<dyn NavCallback>>
```

Comment:
The navigation handler should live in `Webview` and not in `WebContextDat`, same as `ipc_handler`.